src/contracts/proxies/actions/BasicActions.sol

```diff
@@ -276,21 +239,17 @@ contract BasicActions is CommonActions, IBasicActions {
     uint256 _deltaWad
   ) external delegateCall {
     // Unlocks token amount from the SAFE
-    _modifySAFECollateralization(_manager, _safeId, -_deltaWad.toInt(), 0);
+    ODSafeManager.SAFEData memory _safeInfo = ODSafeManager(_manager).safeData(_safeId);
+
+    _modifySAFECollateralization(_manager, _safeId, -_deltaWad.toInt(), 0, true);
     // Transfers token amount to the user's address
     _collectAndExitCollateral(_manager, _collateralJoin, _safeId, _deltaWad);
   }
 
   /// @inheritdoc IBasicActions
-  function repayAllDebt(
-    address _manager,
-    address _taxCollector,
-    address _coinJoin,
-    uint256 _safeId
-  ) external delegateCall {
+  function repayAllDebt(address _manager, address _coinJoin, uint256 _safeId) external delegateCall {
     address _safeEngine = ODSafeManager(_manager).safeEngine();
     ODSafeManager.SAFEData memory _safeInfo = ODSafeManager(_manager).safeData(_safeId);
-    ITaxCollector(_taxCollector).taxSingle(_safeInfo.collateralType);
 
```

Comment:
is there still a tax here since the call is going directly to SafeENGINE?

Reply:
we are still executing the call via the ODSafeManager here, so it will still be taxed - if we were executing the call directly to SAFEEngine, then there would be no tax.